app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -25,19 +26,61 @@ function getIconTypeFromKind(kind: string) {
   return kindIconMap[getKindName(kind)] ?? 'placeholder';
 }
 
-export const ManagedResourceObject = ({
-  resource: {
+const resourceKindParsingRegex = /(?<provider>\w+)\/(?<resourceType>[\w-]+)@(?<version>\w+)/;
+const resourceNameParsingRegex = /:([\w-]+)$/;
+
+function useResourceLink(resource: IManagedResourceSummary): string | null {
+  const resourceType = (resource.kind.match(resourceKindParsingRegex) as any)?.groups?.resourceType;
+  const acct = resource.locations.account;
+  const { app, stack, detail } = resource.moniker;
+  const resourceNameMatch = resource.id.match(resourceNameParsingRegex);
+  const resourceName = resourceNameMatch ? resourceNameMatch[1] : '';
+
+  switch (resourceType) {
+    case 'cluster':
+      return useSref('home.applications.application.insight.clusters', {
+        acct,
+        application: app,
+        detail,
+        stack,
+        q: resourceName,
+      })?.href;
+
+    case 'security-group':
+      return useSref('home.applications.application.insight.firewalls', {
+        acct,
+        application: app,
+        detail,
+        stack,
+        q: resourceName,
+      })?.href;
+
+    case 'classic-load-balancer':
+    case 'application-load-balancer':
+      return useSref('home.applications.application.insight.loadBalancers', {
+        acct,
+        application: app,
+        detail,
+        stack,
+        q: resourceName,
+      })?.href;
+  }
+  return null;
+}
+
+export const ManagedResourceObject = ({ resource, artifact, depth }: IManagedResourceObjectProps) => {
+  const { version: currentVersion, buildNumber: currentBuild } = parseName(artifact?.versions.current || '') || {};
+
+  const {
     kind,
     moniker: { app, stack, detail },
-  },
-  artifact,
-  depth,
-}: IManagedResourceObjectProps) => {
-  const { version: currentVersion, buildNumber: currentBuild } = parseName(artifact?.versions.current || '') || {};
+  } = resource;
+
   return (
     <ObjectRow
       icon={getIconTypeFromKind(kind)}
       title={[app, stack, detail].filter(Boolean).join('-')}
+      link={useResourceLink(resource)}
```

Comment:
calling this function calls hooks, which ideally should all get pulled to the very start of the component to avoid confusion (e.g. if a conditional or other non-deterministic bit of logic gets added later)

Reply:
Yea, the build failure was related to the conditional call made inside the `useResourceLink` function. I'm moving it to the form you described.